### PR TITLE
Always start remote state as 'loading'

### PR DESCRIPTION
**Pull request by AlabasterAxe**
This aligns the status strategies between local and remote. The idea here is that remote starts as 'loading' and the local store is either elected leader and actually starts loading the data or, is not elected leader and explicitly designates its read status as offline.

The context here is that if we're connected to the remote we want to wait until the loading is done but if we're offline, we want to show the project as soon as the local load is done. This works by the remote explicitly reporting offline for read in the case that it's unable to connect to the remote or if it detects an offline state some other way.

### CODE REVIEW

File: packages/trimerge-sync/src/types.ts [hunk 1]
```diff
@@ -64,7 +64,10 @@ export type LocalSaveStatus =
   | 'pending' /** changes in local memory, not sent to store yet */
   | 'saving'; /** sent changes to local store, no `ack` yet */
 
-export type RemoteConnectStatus = 'offline' | 'connecting' | 'online';
+export type RemoteConnectStatus =
+  | 'offline' // this indicates that we've attempted to start the remote but it has somehow gone offline
+  | 'connecting' // this indicates that we've started to connect to the remote
+  | 'online'; // thes indicates we've connected to the remote
```
Comment (AlabasterAxe) [inline packages/trimerge-sync/src/types.ts:70]:
```suggestion
  | 'offline' 
  | 'connecting' 
  | 'online';
```

**Comment by AlabasterAxe** [comment]
There is one more case that I want to update here which is to explicitly set remote state to offline if getremote is unset.

**Comment by AlabasterAxe** [comment]
BTW, was going to write a test for the other uncovered line but it feels a bit contrived. The only way I'm aware that would happen was if the store was shut down but it received a leader election event? I'll leave that uncovered unless you have strong objections.

**Comment by AlabasterAxe** [comment]
@marcello3d if you have a second to approve this. I think all the updated snapshots make sense.

**Review by marcello3d — APPROVED** [review]

**Pull request merged**